### PR TITLE
DSS API specification cleanup & update

### PR DESCRIPTION
**Pull request by bskinner**
## Context

* Cleanup of discriminator use
* Cleanup of some date/time format descriptions
* Adds more `format` and `description` elements
* Removes `speccy` and switches to [redocly/openapi-cli](https://github.com/Redocly/openapi-cli)
* Updates summaries to fix linter errors, and prevent operation ids from appearing in the index column
* Removed unreferenced types & schemas

Mailing Address Before & After

Cc. @yufengwng @ssettipalli 

## FUD Score

- [x] :relaxed: All good, business as usual

## How do we demo these changes?

in `docs/specification` run `npm run start`. The docs can then be accessed at `http://127.0.0.1:8080`

## Testing

- [x] I have written zero automated tests but have poked around locally to verify proper functionality

## Release

- [x] These changes require no special release procedures--just code!

### CODE REVIEW

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 1]
```diff
@@ -57,15 +59,17 @@ JoinMailingListPayload:
       description: The umbrella id of the mailing list the user is interested in (one of umbrellaId OR studyGuid but not both is required)
       example: cmi
       type: string
+      format: guid
     studyGuid:
       description: The studyGuid of the mailing list the user is interested in (one of umbrellaId OR studyGuid but not both is required)
       example: ANGIO
       type: string
+      format: guid
```
Comment (MocanaAtBroad) [inline pepper-apis/docs/specification/src/components/schemas.yml:60]: Is this `guid` format something that is defined somewhere (in this doc or is a well-known thing)?

**Review by MocanaAtBroad — COMMENTED** [review]
I see why you wanted to upgrade! A bunch of stuff got cleaned up, particularly those refs. discriminators are also a nice addition.
Looks good overall and redoc and related scripts seem to handle new spec just fine.
Have one question.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 1 above)
Comment (bskinner) [reply pepper-apis/docs/specification/src/components/schemas.yml:60]: Nope, it's not- the list of supported formats are defined [here](https://swagger.io/specification/#data-type-format). If the value of the `format` property isn't one of the known values, tools are supposed to treat it as a open-string valued property that can have any value.

From my understanding of the platform, the definition of `GUID` is pulling double-duty- as a specific set of types (Study GUID, Answer GUID, etc), and also as a general way of identifying a unique identifier type. This is using the latter definition, and my intent here is to more visibly call out the identifier types in the API outside of just their name.

**Comment by bskinner** [comment]
It feels quite nice 😂- those `$refs` and the `discriminator` weirdness have been bugging me from the beginning.

> I see why you wanted to upgrade! A bunch of stuff got cleaned up, particularly those refs. discriminators are also a nice addition.
> Looks good overall and redoc and related scripts seem to handle new spec just fine.
> Have one question.

**Comment by bskinner** [comment]
This is ready for review!

File: pepper-apis/docs/specification/src/endpoints/user.profile.address.temp.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@ put:
 
 delete:
   operationId: deleteTemporaryMailingAddress
-  description: removed the temporary mailing address for a participant
+  summary: Delete a patient's temporary mailing address
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/endpoints/user.profile.address.temp.yml]:
```suggestion
  summary: Delete a participant's temporary mailing address
```

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 3]
```diff
@@ -2117,24 +2171,27 @@ Invitation:
...
       nullable: true
       example: 2020-05-13T18:58:33.887Z
+      format: YYYY-MM-DDThhmmss.sssZ (ISO-8601)
     acceptedAt:
-      description: timestamp of when invitation was accepted, in ISO-8601 UTC format
+      description: timestamp of when invitation was accepted.
       type: string
       nullable: true
       example: 2020-05-13T19:58:33.887Z
+      format: YYYY-MM-DDThhmmss.sssZ (ISO-8601)
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/components/schemas.yml]: Do we need colon separator in the time portion of the format?

**Review by yufengwng — COMMENTED** [review]
Looks great!

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 3 above)
Comment (bskinner) [reply pepper-apis/docs/specification/src/components/schemas.yml]: Ooo, good catch. Yes, we do- we're using the extended format, not the basic. Thank you!

Comment (bskinner) [reply pepper-apis/docs/specification/src/components/schemas.yml]: Fixed in 829b72f48950963b66cc03e82b6b03020ae5fe80

**Review by yufengwng — DISMISSED** [review]
👍

**Comment by bskinner** [comment]
Brought the branch up to date with `develop`.

**Review by yufengwng — APPROVED** [review]

**Pull request merged**